go/vt/vtgate/vindexes/vschema.go

```diff
@@ -623,6 +653,48 @@ func (vschema *VSchema) FindTableOrVindex(keyspace, name string, tabletType topo
 	return nil, nil, NotFoundError{TableName: name}
 }
 
+func (vschema *VSchema) FindView(keyspace, name string) sqlparser.SelectStatement {
+	if keyspace == "" {
+		switch {
+		case len(vschema.Keyspaces) == 1:
+			for _, schema := range vschema.Keyspaces {
+				keyspace = schema.Keyspace.Name
+				break
+			}
+		default:
+			ks, ok := vschema.uniqueTables[name]
+			if !ok {
+				return nil
+			}
+			if ok && ks == nil {
+				return nil
+			}
+			keyspace = *ks
+		}
+	}
+
+	ks := vschema.Keyspaces[keyspace]
+	if ks == nil {
+		return nil
+	}
+
+	statement, ok := ks.Views[name]
+	if !ok {
+		return nil
+	}
+
+	// We do this to make sure there is no shared state between uses of this AST
+	statement = sqlparser.CloneSelectStatement(statement)
+	sqlparser.Rewrite(statement, func(cursor *sqlparser.Cursor) bool {
+		col, ok := cursor.Node().(*sqlparser.ColName)
+		if ok {
+			cursor.Replace(sqlparser.NewColNameWithQualifier(col.Name.String(), col.Qualifier))
+		}
+		return true
+	}, nil)
```

Comment:
could you explain why this rewriting is needed here

Reply:
you mean add a comment to it, right? the why is because of the stupid `Metadata` field in this stuct already being filled in.

Reply:
What do you think about changing this rewriting to happen once when the vschema is loaded, instead of on each call to `FindView`? The rewriting does not seem to be dependent on `keyspace` or `name`.